src/Api/PubnubApi/Pubnub.cs

```diff
@@ -557,12 +577,23 @@ public bool Reconnect<T>()
         public bool Reconnect<T>(bool resetSubscribeTimetoken)
         {
             bool ret = false;
-            if (savedSubscribeOperation is EndPoint.SubscribeOperation<T>)
+            if (pubnubConfig[InstanceId].EnableEventEngine)
+            {
+                if (subscribeEventEngineFactory.HasEventEngine(InstanceId))
+                {
+                    var subscribeEventEngine = subscribeEventEngineFactory.GetEventEngine(InstanceId);
+                    subscribeEventEngine.EventQueue.Enqueue(new ReconnectEvent() { Channels = (subscribeEventEngine.CurrentState as SubscriptionState).Channels, ChannelGroups = (subscribeEventEngine.CurrentState as SubscriptionState).ChannelGroups, Cursor = resetSubscribeTimetoken ? null : (subscribeEventEngine.CurrentState as SubscriptionState).Cursor });
```

Comment:
This line should be splitted - it's very hard to understand while reading ;/

Reply:
I cross checked it.. In IDE it looks pretty.

Reply:
Still, Did some indentation

Reply:
indented